Remove pywerview dependency

mpgn opened the pull request:
## Description

The age of pywerview is over, the time of impacket has come

## Type of change
Please delete options that are not relevant.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Tester c'est douter

## Screenshots (if appropriate):

![image](https://github.com/user-attachments/assets/3cf5778a-f6a5-4c63-955a-d6d19fa32ad3)

![image](https://github.com/user-attachments/assets/11e5d43d-bf48-4047-8106-1fd656630518)


## Checklist:

- [x] I have ran Ruff against my changes (via poetry: `poetry run python -m ruff check . --preview`, use `--fix` to automatically fix what it can)
- [x] I have added or updated the tests/e2e_commands.txt file if necessary
- [x] New and existing e2e tests pass locally with my changes
- [x] My code follows the style guidelines of this project (should be covered by Ruff above)
- [x] If reliant on third party dependencies, such as Impacket, dploot, lsassy, etc, I have linked the relevant PRs in those projects
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (PR here: https://github.com/Pennyw0rth/NetExec-Wiki)

NeffIsBack commented:
Loggedon-users finally working with kerberos as well:
![image](https://github.com/user-attachments/assets/06984a16-388c-42ad-a0c3-3ad63e71a0bf)

NeffIsBack commented:
Imo we should merge `--loggedon-users-filter` with `--loggedon-users` similar to all the other flags work, e.g. `--users`: if `--loggedon-users` has no user specified, return all. If you specify `--loggedon-users username` this user is filtered.

Thoughts?

NeffIsBack commented on nxc/protocols/smb.py:
```diff
@@ -1241,113 +1239,42 @@ def get_dc_ips(self):
-                    queried_groupname=self.args.local_groups,
-                    list_groups=bool(not self.args.local_groups),
-                    recurse=False,
-                )
+        
+        self.logger.display("Trying with SAMRPC protocol")
+        groups = SamrFunc(self).get_local_groups()
+        if groups:
+            self.logger.success("Enumerated local groups")
+            self.logger.debug(f"Local groups: {groups}")
+
+        for group_name, group_rid in groups.items():
+            self.logger.highlight(f"{group_rid} - {group_name}")
+            group_id = self.db.add_group(self.hostname, group_name, rid=group_rid)[0]
+            self.logger.debug(f"Added group, returned id: {group_id}")
```
Would be nice if we could have the possibility to specify a group which is then queried and all users are displayed. I think that was originally also possible with the old one (although i think it is broken in the main branch)

![image](https://github.com/user-attachments/assets/01288a6c-f17e-4345-ac7a-2d921a7ab97d)

NeffIsBack commented:
Ported the functionality to enumerate groupmembers by specifying the group in `--groups` from the smb protocol to the ldap protocol:
![image](https://github.com/user-attachments/assets/bf7a238b-3c16-4db4-b8b2-42fcd47d457f)

Also deprecated the group-mem module, because it does the exact same thing.

NeffIsBack commented:
`--computers` now also ported to ldap as well
![image](https://github.com/user-attachments/assets/5c11befc-eaed-41d3-b993-d95660e13465)

mpgn commented:
This pull request includes several changes across multiple files to deprecate certain functionalities, add new features, and remove dependencies. The most important changes are summarized below:

### Deprecations:
* [`nxc/modules/group-mem.py`](diffhunk://#diff-999ceeb478b20a36c962e12abce0132d9d6ebc1076a79c06fcc46e9725693c65L22-R24): Deprecated the `group-mem` module, advising users to use the ldap flag `--groups "Administrators"` instead. Added deprecation warnings in the `options` and `on_login` methods. [[1]](diffhunk://#diff-999ceeb478b20a36c962e12abce0132d9d6ebc1076a79c06fcc46e9725693c65L22-R24) [[2]](diffhunk://#diff-999ceeb478b20a36c962e12abce0132d9d6ebc1076a79c06fcc46e9725693c65R39-R41)
* [`nxc/protocols/smb.py`](diffhunk://#diff-03a263aab56e8880814c8a586230088900d62ab9c53b93744492f6fcd1bf7630L1244-R1303): Deprecated several methods (`smb_sessions`, `disks`, `local_groups`, `groups`, `computers`, `loggedon_users`) and advised users to use alternative options or protocols.

### New Features:
* [`nxc/protocols/ldap.py`](diffhunk://#diff-58dbbb4ad45d7ca196b2c60c427281102b0aad1dade02459487c9c598d144e76R677-R717): Added a new method `computers` to enumerate domain computers. Updated the `groups` method to handle specific group enumeration and added debug logging.
* [`nxc/protocols/ldap/proto_args.py`](diffhunk://#diff-b24f079d1a9c675e6aed6fcc4ded0e1906929dee6a7f7b447692c67175950befL25-R26): Added new command-line arguments `--computers` and updated `--groups` to allow specifying a group for member enumeration.

### Dependency Removal:
* [`pyproject.toml`](diffhunk://#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L62): Removed the `pywerview` dependency due to its requirement for `libkrb5-dev` which is not default on Kali Linux.

### Miscellaneous:
* [`nxc/protocols/smb/samrfunc.py`](diffhunk://#diff-69f9a9fe451c9dd692e56f0311615ee107fcacbf00fc18ba878bfe1d4e950133R29): Added the `host` attribute to the `SamrFunc` class initialization.
* [`nxc/protocols/smb.py`](diffhunk://#diff-03a263aab56e8880814c8a586230088900d62ab9c53b93744492f6fcd1bf7630L17-R17): Added imports for `srvs` and `wkst` from `impacket.dcerpc.v5`.

NeffIsBack commented:
> Imo we should merge `--loggedon-users-filter` with `--loggedon-users` similar to all the other flags work, e.g. `--users`: if `--loggedon-users` has no user specified, return all. If you specify `--loggedon-users username` this user is filtered.

Done:
![image](https://github.com/user-attachments/assets/90d6b6d9-e32f-428e-91c1-37ebc7250a7c)

mpgn commented:
Not tested very well but should work 🎉 

![image](https://github.com/user-attachments/assets/e766dbb1-8c07-4190-b3a2-51e7a8bf463d)

NeffIsBack commented:
> Not tested very well but should work 🎉

Added exception handling for `rpc_access_denied`, the rest works flawlessly! Nice one!

NeffIsBack reviewed (approved):
LGTM

ThePirateWhoSmellsOfSunflowers commented:
:sob:
